apps/src/lib/util/firehose.js

```diff
@@ -375,12 +375,18 @@ function getSingleton() {
 }
 
 function putRecord(data, options) {
+  if (IN_UNIT_TEST) {
```

Comment:
i think we're moving away from using `IN_UNIT_TEST` to change behaviors -- can we globally stub these functions instead? i'm not sure of the best place for this, but it would be similar to the way we fail on console errors/warnings on [unit-tests.js](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/test/unit-tests.js), and there are somewhat similar mocks/stubs in [apps/test/util/](https://github.com/code-dot-org/code-dot-org/tree/staging/apps/test/util) (e.g., stubFirehose.js or stubRailsAuthenticityToken.js)

Reply:
ah, i was not aware of that. unfortunately trying to do this with stubs gets tricky because a lot of our tests already stub `putRecord` to unit test reporting. probably not insurmountable, but not something i have time to take on right now. 

if you don't think we should merge this i can just continue applying this change from my local stash when needed, but i thought it would be nice to share my local QOL improvement with the team!

Reply:
that's understandable. i have some thoughts / stuff i found that might help if we do want to ship this change in the future--

we already have a [stubFirehose()](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/test/util/stubFirehose.js) method used in a [similar way for integration tests](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/test/integration-tests.js#L16). i think we'd need to add the ability to stub firehose beforeEach/afterEach (the existing implementation stubs it in before/after, which we wouldn't want for unit tests).

i'd have to mess around with the individual tests that already stub firehose, but i wanted to share that info because i didn't know it existed before today!